pkg/kubernetes/database_cluster.go

```diff
@@ -29,27 +30,39 @@ import (
 
 // ListDatabaseClusters returns list of managed database clusters.
 func (k *Kubernetes) ListDatabaseClusters(ctx context.Context, namespace string) (*everestv1alpha1.DatabaseClusterList, error) {
-	return k.client.ListDatabaseClusters(ctx, namespace, metav1.ListOptions{})
+	result := &everestv1alpha1.DatabaseClusterList{}
+	err := k.k8sClient.List(ctx, result, &ctrlclient.ListOptions{Namespace: namespace})
+	return result, err
 }
 
 // GetDatabaseCluster returns database clusters by provided name.
 func (k *Kubernetes) GetDatabaseCluster(ctx context.Context, namespace, name string) (*everestv1alpha1.DatabaseCluster, error) {
-	return k.client.GetDatabaseCluster(ctx, namespace, name)
+	result := &everestv1alpha1.DatabaseCluster{}
+	err := k.k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: name}, result)
```

Comment:
`k.k8sClient.Get` and `k.k8sClient.List` operations use cache that is being updated in background and in case resource in K8S is changed